pkg/bmc/errors.go

```diff
@@ -26,3 +26,15 @@ func (e CredentialsValidationError) Error() string {
 	return fmt.Sprintf("Validation error with BMC credentials: %s",
 		e.message)
 }
+
+// BootInterfaceValidationError is returned when the BMC can't validate
+// the BootInterface
+// (e.g veirfy if the boot interface has support for virtual media)
+type BootInterfaceValidationError struct {
+	message string
+}
+
+func (e BootInterfaceValidationError) Error() string {
```

Comment:
I'd prefer to have a generic `Validate` function in case we need to validate something that is not boot related.

Reply:
You mean have a generic Error ? (and change BootInterfaceValidationError to ValidateError?)

Reply:
Yep, s/BootInterface// in both

Reply:
Do we need a special error type at all? Do we expect someone to catch this error and do something unique with it that they would not do with another type of error?

Reply:
hummm, it should be fine to move the node to error state  so we don't need a specific Error.

Reply:
So maybe we can use `error.New()` or `fmt.Errorf()` inline?

Reply:
Sounds good!